libs/output-mapping/src/Writer/TableWriter.php

```diff
@@ -194,7 +203,21 @@ private function createLoadTableTask(
         // some scenarios are not supported by the SAPI, so we need to take care of them manually here
         // - columns in config + headless CSV (SAPI always expect to have a header in CSV)
         // - sliced files
-        if (!$destinationTableExists && $hasColumns) {
+        if ($typedTableEnabled && !$destinationTableExists) {
+            $tableDefinition = (new TableDefinition(
+                $systemMetadata['componentId'],
+                $destinationBucketDetails['backend'])
+            )->setName($destination->getTableName());
+            $tableDefinition->setPrimaryKeysNames(
+                PrimaryKeyHelper::normalizeKeyArray($this->logger, $config['primary_key'])
+            );
+            foreach ($config['column_metadata'] as $columnName => $metadata) {
```

Comment:
this should go into the create table definition method

Reply:
yup, I'm almost finished with my refactoring, using factories